### PR TITLE
porting st7735 tft lcd to diy2400 e28 target

### PR DESCRIPTION
**Pull request by gcrisis**
I find that st7735 tft lcd has some  different types.They have different setup,like "TFT_RGB_ORDER" and "ST7735_GREENTAB160x80" or "ST7735_REDTAB160x80";
If one buy a lcd not match the default setup,the screen will not display graphics correctly.
I don't know how to make a one for all setup,so I keep origin setup.

https://user-images.githubusercontent.com/13515569/151187829-c8885b61-ea05-477e-b0e7-d82e638d60e7.mp4

### CODE REVIEW

File: src/include/target/DIY_2400_TX_ESP32_SX1280_E28.h [hunk 1]
```diff
@@ -28,15 +30,29 @@
...
 #define GPIO_PIN_RCSIGNAL_RX    13
 #define GPIO_PIN_RCSIGNAL_TX    13
 #define GPIO_PIN_LED_WS2812     15
 #define GPIO_PIN_FAN_EN         17
 
+#ifdef HAS_FIVE_WAY_BUTTON
+#define GPIO_PIN_FIVE_WAY_INPUT1     39
+#define GPIO_PIN_FIVE_WAY_INPUT2     35
+#define GPIO_PIN_FIVE_WAY_INPUT3     34
+#endif
```
Comment (FOG-Yamato) [inline src/include/target/DIY_2400_TX_ESP32_SX1280_E28.h:50]: This is not the recommended way to wire up a 5 way joystick, the better option is to use a voltage divider on a single ADC pin.

File: src/include/target/DIY_2400_TX_ESP32_SX1280_E28.h [hunk 2]
```diff
@@ -28,15 +30,29 @@
...
 // Output Power
 #define MinPower            PWR_10mW
 #define MaxPower            PWR_250mW
 #define POWER_OUTPUT_VALUES {-15,-11,-8,-5,-1}
 
 #define Regulatory_Domain_ISM_2400 1
+#define TARGET_DIY_2400_TX_ESP32_SX1280_E28
```
Comment (FOG-Yamato) [inline src/include/target/DIY_2400_TX_ESP32_SX1280_E28.h:57]: This is already defined in the PlatformIO targets, doesn't need to be present here as well.

File: src/lib/SCREEN/TFT/TFT_eSPI_User_Setup.h [hunk 3]
```diff
@@ -195,12 +195,14 @@
 
 // For ESP32 Dev board (only tested with ILI9341 display)
 // The hardware SPI can be mapped to any pins
+#ifdef HAS_TFT_SCREEN
```
Comment (FOG-Yamato) [inline src/lib/SCREEN/TFT/TFT_eSPI_User_Setup.h:198]: It would be better if it encompassed all the defines in this header file.

**Review by FOG-Yamato — CHANGES_REQUESTED** [review]
I would also make sure that the menus look okay as this was never used outside of the Axis Thor target.

**Comment by JyeSmith** [comment]
@gcrisis looks like this needs a rebase and address some comments.  Any updates?

**Pull request closed without merging**